data/batch.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import dataclass
+
+import torch
+
+from modules.encoderdecoder import subsequent_mask
+
+
+@dataclass
+class Record:
+    src: torch.Tensor
+    tgt: torch.Tensor
+    src_mask: torch.Tensor
+
+
+class Batch:
```

Comment:
This class is following us from the annotated transformer blogpost, but I think it's not really necessary. The functionality it provides should rather be a part of the dataset logic, so you could get the same structures by doing `dataset[idx]`, without any batching logic invoved.

But I think we can wait with this change, and do some refactoring after we're done with reviewing "translation" experiments.